changelogs/unreleased/gh-8516-luajit-fixes.md

```diff
@@ -4,3 +4,10 @@ Backported patches from the vanilla LuaJIT trunk (gh-8516). The following issues
 were fixed as part of this activity:
 
 * Fixed `IR_LREF` assembling for the GC64 mode on x86_64.
+* Fix canonicalization of +-0.0 keys for `IR_NEWREF`.
+* Fixed result truncation for `bit.rol` on x86_64 platforms.
+* Fixed saved bytecode encapsulated in ELF objects.
+* Fixed `lua_yield()` invocation inside C hooks.
+* Fixed memory chunk allocation beyond the memory limit.
+* Fix TNEW load forwarding with instable types.
```

Comment:
```suggestion
* Fixed canonicalization of +-0.0 keys for `IR_NEWREF`.
* Fixed result truncation for `bit.rol` on x86_64 platforms.
* Fixed saved bytecode encapsulated in ELF objects.
* Fixed `lua_yield()` invocation inside C hooks.
* Fixed memory chunk allocation beyond the memory limit.
* Fixed TNEW load forwarding with instable types.
```

Reply:
@andreyaksenov, thanks for your review! Applied all your suggestions.